homeassistant/components/legrandinone.py

```diff
@@ -0,0 +1,505 @@
+CONF_DEVICE_ID = 'legrand_id'
+CONF_DEVICE_TYPE = 'device_type'
+CONF_DEVICE_UNIT = 'device_unit'
+CONF_COMM_MEDIA = 'media'
+CONF_COMM_MODE = 'comm_mode'
+CONF_DEVICES = 'devices'
+CONF_AUTOMATIC_ADD = 'automatic_add'
+CONF_FIRE_EVENT = 'fire_event'
+CONF_MEDIA = 'iobl_media'
+CONF_COMM_MODE = 'iobl_comm_mode'
+CONF_PACKET_TYPE = 'iobl_pkt_type'
+CONF_DIMENSION_VALUES = 'iobl_dim_values'
+CONF_IGNORE_DEVICES = 'ignore_devices'
+CONF_RECONNECT_INTERVAL = 'reconnect_interval'
+
+DATA_DEVICE_REGISTER = 'iobl_device_register'
+DATA_ENTITY_LOOKUP = 'iobl_entity_lookup'
+DEFAULT_RECONNECT_INTERVAL = 10
+DEFAULT_SIGNAL_REPETITIONS = 1
+CONNECTION_TIMEOUT = 10
+
+EVENT_BUTTON_PRESSED = 'button_pressed'
+EVENT_KEY_ID = 'legrand_id'
+EVENT_KEY_COMMAND = 'bus_command'
+EVENT_TYPE_COMMAND = 'what'
+
+DEVICE_TYPE_AUTOMATION = 'automation'
+DEVICE_TYPE_LIGHT = 'light'
+
+DOMAIN = 'legrandinone'
+
+SERVICE_SEND_COMMAND = 'send_packet'
+
+SIGNAL_AVAILABILITY = 'iobl_device_available'
+SIGNAL_HANDLE_EVENT = 'iobl_handle_event_{}'
+
+TMP_ENTITY = 'tmp.{}'
+
+DEVICE_DEFAULTS_SCHEMA = vol.Schema({
+    vol.Optional(CONF_FIRE_EVENT, default=False): cv.boolean,
+    vol.Optional(CONF_MEDIA, default='plc'): cv.string,
+    vol.Optional(CONF_COMM_MODE, default='unicast'): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PORT): vol.Any(cv.port, cv.string),
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_RECONNECT_INTERVAL,
+                     default=DEFAULT_RECONNECT_INTERVAL): int,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+SEND_COMMAND_SCHEMA = vol.Schema({
+    vol.Required(CONF_DEVICE_ID): cv.string,
+    vol.Required(CONF_DEVICE_TYPE): cv.string,
+    vol.Required(CONF_DEVICE_UNIT): cv.string,
+    vol.Required(CONF_COMM_MEDIA): cv.string,
+    vol.Required(CONF_COMM_MODE): cv.string,
+    vol.Required(CONF_COMMAND): cv.string,
+    vol.Optional(CONF_PACKET_TYPE): cv.string,
+    vol.Optional(CONF_DIMENSION_VALUES): [cv.string],
+})
+
+
+def identify_event_type(event):
+    """Look at event to determine type of device.
+
+    Async friendly.
+    """
+    if event.get('type') == EVENT_KEY_COMMAND:
+        return EVENT_KEY_COMMAND
+
+    return 'unknown'
+
+
+async def async_setup(hass, config):
+    """Set up the IOBL component."""
+    from iobl.protocol import create_iobl_connection
+    import serial
+
+    # Allow entities to register themselves by device_id to be looked up when
+    # new IOBL events arrive to be handled
+    hass.data[DATA_ENTITY_LOOKUP] = {
+        EVENT_KEY_COMMAND: defaultdict(list),
+    }
+
+    # Allow platform to specify function to register new unknown devices
+    hass.data[DATA_DEVICE_REGISTER] = {}
+    hass.data[DATA_DEVICE_REGISTER][EVENT_KEY_COMMAND] = {}
+
+    async def async_send_command(call):
+        """Send legrandinone command."""
+        _LOGGER.debug('LegrandInOne command for %s', str(call.data))
+
+        data = cast(Dict[str, Any], {
```

Comment:
Why cast? We're not typing the whole module anyway.

Reply:
Fixed that too